File: 3254/solution.java
import java.util.Arrays;

class Solution {
    public int[] resultsArray(int[] nums, int k) {
        int[] res = new int[nums.length - k + 1];
        Arrays.fill(res, -1);
        int streak = 0;
        for (int i = 0; i < nums.length; i++) {
            if (i == 0 || nums[i] == nums[i - 1] + 1) {
                streak++;
            } else {
                streak = 1;
            }
            if (streak >= k) {
                res[i - k + 1] = nums[i];
            }
        }
        return res;
    }
}

// time O(n)
// space O(1)

File: 3254/solution.py
class Solution:
    def resultsArray(self, nums: list[int], k: int) -> list[int]:
        curr = 0
        l = 0
        res = []
        for r in range(len(nums)):
            if r > 0 and nums[r] == nums[r - 1] + 1:
                curr += 1
            if r >= k:
                if nums[l + 1] == nums[l] + 1:
                    curr -= 1
                l += 1
            if r >= k - 1 and curr == k - 1:
                res.append(nums[r])
            elif r >= k - 1:
                res.append(-1)
        return res
